Simplified/AppInfrastructure/NYPLLibraryNavigationController.m

```diff
@@ -11,70 +22,27 @@
 #import "NYPLSettingsPrimaryTableViewController.h"
 #endif
 
-#import "SimplyE-Swift.h"
 
-#if defined(FEATURE_DRM_CONNECTOR)
-#import <ADEPT/ADEPT.h>
-#endif
+@interface NYPLLibraryNavigationController ()
 
-@implementation NYPLMyBooksNavigationController
+@end
 
-#pragma mark NSObject
+@implementation NYPLLibraryNavigationController
 
-- (instancetype)init
-{
-  NYPLMyBooksViewController *viewController =
-  [[NYPLMyBooksViewController alloc] init];
-  
-  self = [super initWithRootViewController:viewController];
-  if(!self) return nil;
-  
-  self.tabBarItem.image = [UIImage imageNamed:@"MyBooks"];
-  
 #ifdef SIMPLYE
-  viewController.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc]
-                                                     initWithImage:[UIImage imageNamed:@"Catalog"] style:(UIBarButtonItemStylePlain)
-                                                     target:self
-                                                     action:@selector(switchLibrary)];
-  viewController.navigationItem.leftBarButtonItem.accessibilityLabel = NSLocalizedString(@"AccessibilitySwitchLibrary", nil);
-  viewController.navigationItem.leftBarButtonItem.enabled = YES;
-#endif
-  
-  [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(currentAccountChanged) name:NSNotification.NYPLCurrentAccountDidChange object:nil];
-
-  return self;
-}
-
-- (void)dealloc
+- (void)setNavigationLeftBarButtonForVC:(UIViewController *)vc
 {
-  [[NSNotificationCenter defaultCenter] removeObserver:self];
+  vc.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc]
+                                         initWithImage:[UIImage imageNamed:@"Catalog"] style:(UIBarButtonItemStylePlain)
+                                         target:self
+                                         action:@selector(switchLibrary)];
+  vc.navigationItem.leftBarButtonItem.accessibilityLabel = NSLocalizedString(@"AccessibilitySwitchLibrary", nil);
 }
 
--(void)viewWillAppear:(BOOL)animated
+// for converting this to Swift, see https://bit.ly/3mM9QoH
+- (void)switchLibrary
 {
-  [super viewWillAppear:animated];
-    
-  NYPLMyBooksViewController *viewController = (NYPLMyBooksViewController *)self.visibleViewController;
-  
-  viewController.navigationItem.title = [AccountsManager shared].currentAccount.name;
-    
-}
-
-- (void)currentAccountChanged
-{
-  if (![NSThread isMainThread]) {
-    dispatch_async(dispatch_get_main_queue(), ^{
-      [self popToRootViewControllerAnimated:NO];
-    });
-  } else {
-    [self popToRootViewControllerAnimated:NO];
-  }
-}
-
-#ifdef SIMPLYE
-- (void) switchLibrary
```

Comment:
this entire function was removed from NYPLCatalogNavigationController, NYPLMyBooksNavigationController, NYPLHoldsNavigationController and moved it into their new superclass, NYPLLibraryNavigationController.